### PR TITLE
Edjroz/test/update contract abstract account

### PR DESCRIPTION
**Pull request by ash-burnt**

### CODE REVIEW

File: integration_tests/abstract_account_test.go [hunk 1]
```diff
@@ -182,12 +199,18 @@ func TestXionAbstractAccount(t *testing.T) {
 	)
 	require.NoError(t, err)
 
-	updatedContractstate, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"pubkey":{}}`))
+	updatedContractstate, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"authenticator_by_i_d":{ "id": 0 }}`))
```
Comment (mvid) [inline integration_tests/abstract_account_test.go:202]: does this work with the key misspelling?

File: integration_tests/abstract_account_test.go [hunk 2]
```diff
@@ -21,6 +22,8 @@ import (
 	aatypes "github.com/larry0x/abstract-account/x/abstractaccount/types"
 )
 
+type jsonauthenticator map[string]map[string]string
```
Comment (mvid) [inline integration_tests/abstract_account_test.go:25]: Nit, follow go casing conventions

File: integration_tests/abstract_account_test.go [hunk 3]
```diff
@@ -84,39 +94,46 @@ func TestXionAbstractAccount(t *testing.T) {
...
 	require.Equal(t, uint64(100000), uint64(contractBalance))
 
-	/*
-			NOTE: Ideally we would use this metod, however the QueryContract formats the string making it harder to predict.
-		var ContractResponse interface{}
-			require.NoError(t, xion.QueryContract(ctx, aaContractAddr, fmt.Sprintf(`{"pubkey":{}}`), ContractResponse))
-	*/
-
-	contractState, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"pubkey":{}}`))
+	contractState, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"authenticator_by_i_d":{ "id": 0 }}`))
```
Comment (mvid) [inline integration_tests/abstract_account_test.go:127]: Does this work with "authenticator_by_i_d" vs "authenticator_by_id"

**Review by mvid — APPROVED** [review]

File: integration_tests/abstract_account_test.go (hunk 1 above)
Comment (edjroz) [reply integration_tests/abstract_account_test.go:202]: The contract is expecting the key like this `authenticator_by_i_d` becuase the type in the contract is `AuthenticatorByID`.

File: integration_tests/abstract_account_test.go (hunk 3 above)
Comment (edjroz) [reply integration_tests/abstract_account_test.go:127]: The contract is expecting the key like this `authenticator_by_i_d` becuase the type in the contract is `AuthenticatorByID`.

File: integration_tests/abstract_account_test.go (hunk 2 above)
Comment (edjroz) [reply integration_tests/abstract_account_test.go]: fixed in the latest change

**Pull request merged**